src/public/interfaces.ts

```diff
@@ -527,12 +527,21 @@ export interface SdkError {
   errorCode: ErrorCode;
   /**
   Optional description for the error. This may contain useful information for web-app developers.
-  This string will not be localized and is not for end-user consumption.
+  This string will not be localized and is not for end-user consumption. 
+  App should not dependend on the string content. The exact value may change. This is only for debugging purposes.
   */
   message?: string;
 }
 
 export const enum ErrorCode {
+  /**
+   * API not supported in the current platform.
+   */
+  NOT_SUPPORTED_ON_PLATFORM = 100,
+  /**
+   * Internal error encountered while performing the required operation.
```

Comment:
Do we have a scenario where we are passing it from the platform code ? 

If the scenario/error is not captured by existing codes, could we define more specific error instead of INTERNAL_ERROR as this gives very little information to developer regarding what went wrong in the handling of the API. Also, dev wouldn't know if the error can be rectified by the user and whether he can retry the operation etc.

Reply:
We should keep either internal error or generic error because the exact cause of all errors are not likely to be known

Reply:
Yes. The intention here is to let end user know something went wrong, knowing details of which may not make sense to him. Error can contain message which will contain debug information for dev. Keeping this as INTERNAL_ERROR